sacred/config/custom_containers.py

```diff
@@ -212,10 +213,12 @@ class ReadOnlyList(ReadOnlyContainer, list):
     __setitem__ = ReadOnlyContainer._readonly
     __delitem__ = ReadOnlyContainer._readonly
 
-    def __init__(self, *iterable, message=None, **kwargs):
-        if message is None:
-            message = "This ReadOnlyList is read-only!"
-        super().__init__(*iterable, message=message, **kwargs)
+    def __init__(self, lst, message=None):
```

Comment:
Not sure whether I like the "simpler" `__init__`s. These containers were designed to be compatible with `dict` and `list` with one additional keyword-only argument for the message. But since these containers are not meant to be constructed outside of `make_read_only`, I think that this incompatibility is fine.

Reply:
I think I changed these arguments because I couldn't figure out at the time I couldn't figure out how to use `__reduce__` to set `self.message` using a keyword argument.

Reply:
1. One way to pass keyword arguments seems to be having `ReadOnlyContainer.__reduce__` to return a wrapped Callable instead of `self._class`, as [explained here](https://stackoverflow.com/questions/54802258/pickling-keyword-arguments-in-python-3#comment96381387_54802258).

2. A simpler way to set `self.message` (bypassing the inability to pass keyword arguments) is to pass a dict of attributes as the third return value in `__reduce__`.

![image](https://user-images.githubusercontent.com/1750835/96168538-bdcbb900-0ed5-11eb-9be0-470c6003324a.png)


If you'd like this, I can revert to the old `__init__` arguments and use solution 2.

Reply:
Do we actually need the message keyword? It is currently only used in one place and always set to the same value. I think we can remove it and hard-code the message

Reply:
Removing it sounds good.

Reply:
Removed.

Reply:
Can you also fix the call of `make_read_only` in `initialize.py`?

Reply:
fixed.